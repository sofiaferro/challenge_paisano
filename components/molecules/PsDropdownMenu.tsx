import React from 'react';

const PsDropdownMenu = () => {
  return <div>PsDropdownMenu</div>;
};

export default PsDropdownMenu;
