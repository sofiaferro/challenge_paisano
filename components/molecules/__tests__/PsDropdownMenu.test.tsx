import React from 'react';
import renderer from 'react-test-renderer';

import { describe, expect, it } from '@jest/globals';

import PsDropdownMenu from '../PsDropdownMenu';

const options = [
  { text: 'opt 1', value: 1 },
  { text: 'opt 2', value: 2 },
  { text: 'opt 3', value: 3 },
];

describe('PsDropdownMenu', () => {
  it('renders the component', () => {
    const mockCallback = jest.fn();
    const tree = renderer.create(
      <PsDropdownMenu
        options={options}
        defaultPlaceholder={'test placeholder'}
        onChange={mockCallback}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
