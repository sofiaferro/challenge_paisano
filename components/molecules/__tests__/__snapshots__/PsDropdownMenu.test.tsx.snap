// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PsDropdownMenu renders the component 1`] = `
<div
  className="sc-4b7d43a0-1 eppnxi"
>
  <div
    className="sc-4b7d43a0-0 hwmByI"
    onClick={[Function]}
    style={{}}
  >
    <p>
      test placeholder
    </p>
    <div
      className="sc-4b7d43a0-2 grEoUH"
    >
      <img
        alt="Icon chevron bar"
        data-nimg="1"
        decoding="async"
        height={40}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={40}
      />
    </div>
  </div>
  <ul
    className="sc-4b7d43a0-3 heKFHF"
  >
    <li
      className="sc-4b7d43a0-4 bIdavV"
      onClick={[Function]}
      value={1}
    >
      opt 1
    </li>
    <li
      className="sc-4b7d43a0-4 bIdavV"
      onClick={[Function]}
      value={2}
    >
      opt 2
    </li>
    <li
      className="sc-4b7d43a0-4 bIdavV"
      onClick={[Function]}
      value={3}
    >
      opt 3
    </li>
  </ul>
</div>
`;
